### ISSUE TITLE
Implement AsyncMethodWithoutCancellationToken

### ISSUE THREAD

**Issue by Vannevelj** [Vannevelj/VSDiagnostics#378]
An async method that does not take a `CancellationToken` or `CancellationTokenSource` parameter should show a warning and provide a code fix that implements an optional `CancellationToken` parameter.

From this:

```csharp
async Task MethodAsync(int i, string s) { }
```
To this:
```csharp
async Task MethodAsync(int i, string s, CancellationToken cancellationToken = null) { }
```

Notes:

* Using statements (in case of `async void` there might not be a `System.Threading.x` import yet)
* Methods with varargs or optional parameters
* Existing overloads: one with and one without cancellation token
